src/NAR/Parser.java

```diff
@@ -40,4 +51,45 @@ public static boolean checkUserInput(String statement)
 		Editor.printToConsole("Unknown construct '" + calledConstructName + "' found.");
 		return false;
 	}
+	
+	/**
+	 * This method takes a variable name as input.
+	 * Returns true if the variable has been defined false otherwise.
+	 * @param variable
+	 * @return
+	 */
+	public static boolean isVarDefined(String variable) {
+		return Instance.variables.containsKey(variable);
+	}
+	
+	/**
+	 * This method takes two parameters, a variable name, and it�fs associated number value. 
+	 * It stores the variable name and number together in internal memory. 
+	 * It also can be used to store a new value for an already defined variable.
+	 * @param variable
+	 * @param value
+	 */
+	public static void defineVar(String variable, double value) {
+		Instance.variables.put(variable, value);
+	}
+	
+	/**
+	 * This method takes a variable name as input and 
+	 * returns the numeric value associated with it.
+	 * @param variable
+	 * @return value of variable
+	 * @throws NullPointerException if variable is null
+	 */
+	public static double getVarValue(String variable) throws NullPointerException {
```

Comment:
given that we should always be checking if the variable is defined before calling this function you should probably just throw the exception from inside the getVarValue function and print to the IDE console which variable is not defined. Right now by declaring the variable to throw the exception you are forcing the code using this to handle a NullPointerException which is not really what we want. We want the code using this to completely avoid this exception and only call this function if the variable in question is defined